lib/engine/src/artifact.rs

```diff
@@ -176,6 +176,35 @@ pub trait Artifact: Send + Sync + Upcastable + MemoryUsage {
             .finish_instantiation(trap_handler, &data_initializers)
             .map_err(|trap| InstantiationError::Start(RuntimeError::from_trap(trap)))
     }
+
+    /// Similar to finish_instantiation, except it doesn't invoke start function.
+    unsafe fn finish_instantiation_without_start_func(
```

Comment:
Do we actually need to split `finish_instantiation` in two parts? Can we use `finish_instantiation` directly as a way to postpone `_start` execution?